server/routes/referrals/referralsController.test.ts

```diff
@@ -91,3 +95,87 @@ describe('GET /referrals/:id/form', () => {
     })
   })
 })
+
+describe('GET /referrals/:id/completion-deadline', () => {
+  beforeEach(() => {
+    interventionsService.getDraftReferral.mockResolvedValue({
+      id: '1',
+      completionDeadline: null,
+      serviceCategory: { name: 'accommodation' },
+    })
+  })
+
+  it('renders a form page', async () => {
+    await request(app)
+      .get('/referrals/1/completion-deadline')
+      .expect(200)
+      .expect(res => {
+        expect(res.text).toContain('What date does the accommodation service need to be completed by?')
+      })
+  })
+  // TODO how do we (or indeed, do we) test what happens when the request has a completion deadline - i.e. that the
```

Comment:
can you override the mock for this test and add a completionDealine to it? then check if the date field is populated?

Reply:
The problem isn't getting the data into the view, it's that the testing tools are too limited to be able to check that the fields are populated correctly - all we can really do is a text match on the raw HTML body, but to check the fields we'd need something that parses the HTML and allows us to interrogate its structure. We could introduce something like that into the tests but we wondered whether it's better to leave it for the Cypress tests?

Reply:
fine with me. sounds overly complex to be parsing the html here.